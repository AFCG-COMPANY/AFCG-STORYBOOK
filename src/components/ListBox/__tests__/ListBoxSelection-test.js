import React from 'react';
import { mount } from 'enzyme';
import ListBox from '../';

describe('ListBoxSelection', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      clearSelection: jest.fn(),
      translateWithId: jest.fn(() => 'translation'),
    };
  });

  it('should render', () => {
    const singleSelectionWrapper = mount(<ListBox.Selection {...mockProps} />);
    const multiSelectionWrapper = mount(
      <ListBox.Selection selectionCount={3} {...mockProps} />
    );
    expect(singleSelectionWrapper).toMatchSnapshot();
    expect(multiSelectionWrapper).toMatchSnapshot();
  });

  it('should call `translateWithId` with the id strings needed to translate', () => {
    mount(<ListBox.Selection {...mockProps} />);
    expect(mockProps.translateWithId).toHaveBeenCalledWith('clear.selection');

    mockProps.translateWithId.mockClear();

    mount(<ListBox.Selection {...mockProps} selectionCount={3} />);
    expect(mockProps.translateWithId).toHaveBeenCalledWith('clear.all');
  });
});
