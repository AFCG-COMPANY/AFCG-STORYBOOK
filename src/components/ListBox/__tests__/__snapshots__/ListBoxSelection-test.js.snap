// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxSelection should render 1`] = `
<ListBoxSelection
  clearSelection={[MockFunction]}
  translateWithId={
    [MockFunction] {
      "calls": Array [
        Array [
          "clear.selection",
        ],
        Array [
          "clear.all",
        ],
      ],
    }
  }
>
  <div
    className="bx--list-box__selection"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
    title="translation"
  >
    <Icon
      description="translation"
      fillRule="evenodd"
      focusable="false"
      name="close"
      role="img"
    >
      <svg
        alt="translation"
        aria-label="translation"
        fillRule="evenodd"
        focusable="false"
        height="10"
        name="close"
        role="img"
        viewBox="0 0 10 10"
        width="10"
      >
        <title>
          translation
        </title>
        <path
          d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
          key="key0"
        />
      </svg>
    </Icon>
  </div>
</ListBoxSelection>
`;

exports[`ListBoxSelection should render 2`] = `
<ListBoxSelection
  clearSelection={[MockFunction]}
  selectionCount={3}
  translateWithId={
    [MockFunction] {
      "calls": Array [
        Array [
          "clear.selection",
        ],
        Array [
          "clear.all",
        ],
      ],
    }
  }
>
  <div
    className="bx--list-box__selection bx--list-box__selection--multi"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
    title="translation"
  >
    3
    <Icon
      description="translation"
      fillRule="evenodd"
      focusable="false"
      name="close"
      role="img"
    >
      <svg
        alt="translation"
        aria-label="translation"
        fillRule="evenodd"
        focusable="false"
        height="10"
        name="close"
        role="img"
        viewBox="0 0 10 10"
        width="10"
      >
        <title>
          translation
        </title>
        <path
          d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
          key="key0"
        />
      </svg>
    </Icon>
  </div>
</ListBoxSelection>
`;
